rest/src/main/kotlin/builder/message/modify/MessageModifyBuilder.kt

```diff
@@ -32,20 +33,39 @@ public sealed interface MessageModifyBuilder {
 
     public var attachments: MutableList<DiscordAttachment>?
 
-    public fun addFile(name: String, content: InputStream): NamedFile {
-        val namedFile = NamedFile(name, content)
+    /**
+     * Adds a file with the [name] and [content] to the attachments.
+     */
+    @Deprecated(
+        "Use lazy ChannelProvider instead of InputStream. You should also make sure that the stream/channel is only " +
+                "opened inside the block of the ChannelProvider because it could otherwise be read multiple times " +
+                "(which isn't allowed).",
+        ReplaceWith(
+            "addFile(name, ChannelProvider { content.toByteReadChannel() })",
+            "io.ktor.client.request.forms.ChannelProvider",
+            "io.ktor.utils.io.jvm.javaio.toByteReadChannel",
+        ),
+        DeprecationLevel.WARNING,
+    )
+    public fun addFile(name: String, content: InputStream): NamedFile =
+        addFile(name, ChannelProvider { content.toByteReadChannel() })
+
+    public suspend fun addFile(path: Path): NamedFile =
+        addFile(path.fileName.toString(), ChannelProvider { path.readChannel() })
 
-        files = (files ?: mutableListOf()).also {
-            it.add(namedFile)
+    /**
+     * Adds a file with the [name] and [contentProvider] to the attachments.
+     */
+    public fun addFile(name: String, contentProvider: ChannelProvider): NamedFile =
+        NamedFile(name, contentProvider).also {
+            addFile(it)
         }
 
-        return namedFile
-    }
-
-    public suspend fun addFile(path: Path): NamedFile = withContext(Dispatchers.IO) {
-        addFile(path.fileName.toString(), Files.newInputStream(path))
+    public fun addFile(file: NamedFile) {
+        files = (files ?: mutableListOf()).also {
+            it.add(file)
+        }
     }
```

Comment:
Why did you add this method? It wasn't there before 🤔

Reply:
Was to remove code duplication in the previous implementation, but now it's obsolete cus all other overloads use ChannelProvider. Will remove